### PR TITLE
fix(notifications): fix windows tray icon and notifications

### LINKED ISSUE #4263: [Windows] Status icon in System tray is missing actions for the first launch
The user cannot interact with the Status app icon on Windows System tray.
image

Steps:

1. install status on windows
2. open the app and login
3. in system tray find the app icon and right click it
4. collapse the status app
5. in system tray find the app icon and double click it

**NOTE**: The actions can be seen after the app restarts on Win 11, but the issue persists on Win 10 even after the app relaunch.

**Expected result:**

- right click on the icon brings up context menu with options 
- double click the icon expands the app

**Build**: 0.4.0, master

**OS**: Windows 11 on Parallels VM, Windows 10 real device

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4263

The problem was that there was still the race condition on initiating the Windows Notification manager.

Also, we were not showing notifications if you were already on the channel that received a message, even if the app was minimized. The Notification manager should be the one handling whether to show a notif in that case, not the section module.

### CODE REVIEW

**Review by alexandraB99 — APPROVED** [review]

**Review by saledjenic — APPROVED** [review]
@jrainville seems that the code we had before to delay a bit a creation of an instance of `OsNotification` class is still needed (hope we can improve it in the c++ app). 

> The Notification manager should be the one handling whether to show a notif in that case, not the section module.

I fully agree.

**Review by abhi-status — APPROVED** [review]

**Comment by abhi-status** [comment]
The issue of the user being unable to see the actions by clicking the Status icon in the System Tray,
re-surfaces after a notification is activated --> https://www.loom.com/share/ef625313eafd41b3b9fafb8275699fd2
The first 40sec of the timeline shows the correct desired behaviour. 
After the app the closed (not quit) and a msg is sent, the user cant bring up the app by clicking on the Status icon or see any of the actions.

**Comment by jrainville** [comment]
@abhi-status I investigated the issue and asked @saledjenic for help and we found that it's a very hard issue to fix.

Here is the discussion Sale and I had:
```
Sale
the thing there is not straightforward at all. On mac that's not an issue cause mac handles notification in a separate part of OS what is NSUserNotificationCenter on Win the thing is that notifications are handled via NOTIFYICONDATA which is part of app tray and you can have only one tray registered per app. In our case that is done by Qt (most likely when you call exec of QApplication (or any of inherited classes). Now there is hwid of that tray, but now the thing is that it is not static, but it gets changed over time and there is no chance to hook somehow on that change, so the part which is handling notifications for us most likely after some time (which is ~40sec in the video) points to the wrong hwid and from that moment we lost all tray related functionalities. 
So that issue is very hard to solve. Actually I am not sure how would I solve it except the one (maybe the only way) to patch the Qt source code.
That means we need to extend a bit Qt tray related source code, and build our custom version of Qt.
I am talking about 5.14.2 version.
Maybe once we port to some newer version like Qt6.2 or whatever that notification feature is handled a bit better allowing us to send notification about clicked OS notification bubble back to the app.
I am thinking now maybe (if we decide to patch that in Qt) when we do that we should open a PR on qt repo and suggest that change. 🙂
But that's something we should discuss at least with Iuri before we proceed.
Jo — Aujourd’hui à 09:21
Ok, thanks for your research and input.
Indeed, it seems very hard to fix. Either I had the tray icon working all the time, but the notifications didn't work, or the notifications worked, but the tray icon stopped working after one notification
Sale — Aujourd’hui à 09:22
> but the tray icon stopped working after one notification
most likely because of changed hwid after that
Jo — Aujourd’hui à 09:22
So as far as you know, this is a bug in Qt itself? Like other windows qt qpps would have that same issue?
Sale — Aujourd’hui à 09:23
no, it's not a bug on the Qt side, but that bug started to happen because of an extension we did in order to have some data returned from that OS notification bubble
and we need that in order to know what bubble is clicked to apply appropriate action
like jump to the certain message
but we can updated that in Qt source
and have signal like
messageClicked(string identifier)
that's enough
if we do that we won't have any more issues
```

TLDR we added a patch to QT to receive the notification ID when we click on it, but that made it so that the tray icon bugs when we have new notifications. Also, we checked and QT 6 doesn't fix that.

It's pretty hard to fix and the behaviour we have with this PR is still better than what we had, ie the notifications now work on Windows normally, but the tray icon stops working after the first notification.

I'll open a new issue so that we can track it and fix it later.

**Comment by jrainville** [comment]
I'll merge this PR for now, since it improves the behaviour already. The new issue is here: https://github.com/status-im/status-desktop/issues/4943

**Comment by saledjenic** [comment]
> we added a patch to QT to receive the notification ID when we click on it, but that made it so that the tray icon bugs when we have new notifications

@jrainville just to be clear... what we currently have is that we added OS specific code to deal with OS notifications. On Mac that works fine, on Win that code searches for the registered tray by id and tries to deal with notifications connected to it, but the problem arises when that id gets changed.

Solution:
If we patch a Qt source code, and build our custom Qt framework we will 100% fix the issue and in that case we won't need out OS specific code for handling notifications.
